### PR TITLE
Fix Inside modal container selection issue

### PR DESCRIPTION
**Pull request by anandraj346**
Previously the plugin was writing into the search input text even when the selection is made inside the modal.

**Steps to reproduce:**

1. Open the search and replace modal by clicking the icon on the top left corner.
2. Select some text inside the modal and the search input will be filled with that text i.e., should not happen.

https://github.com/user-attachments/assets/7ad95167-7e26-421b-bbc0-1da8104de3c3

This PR resolves this [issue](https://github.com/badasswp/search-and-replace/issues/9).

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,10 @@
 # Changelog
 
+## 1.2.1
+* Fix in modal selection issue.
+* Fix table content replacement issue.
```
Comment (badasswp) [inline CHANGELOG.md:5]: @anandraj346 Let's remove this and put in a separate PR specifically pointed to `release-1.3.0` branch.

File: src/app.tsx [hunk 2]
```diff
@@ -197,6 +224,124 @@ const SearchReplaceForBlockEditor = () => {
     }
   }
 
+  /**
```
Comment (badasswp) [inline src/app.tsx:227]: @anandraj346 This change with the **Table** block has a lot going on, and should be a `feature` because the previous implementation: `replaceBlockAttribute` did not accurately capture it's edge-cases. Pls let's back out this **Table** block change in this PR and create a new PR for it specifically to be pointed to `release-1.3.0` branch.

**Review by badasswp — CHANGES_REQUESTED** [review]
@anandraj346 Could you create a separate PR for the **Table** block change and point to `release-1.3.0`.

File: src/app.tsx [hunk 3]
```diff
@@ -48,12 +48,35 @@ const SearchReplaceForBlockEditor = () => {
...
+  const inContainer = (selector) => {
+    const selection = window.getSelection();
+    const targetDiv = document.querySelector(selector);
+    
+    if (!selection.rangeCount || !targetDiv) {
+      return false;
+    }
+    
+    const range = selection.getRangeAt(0);
+    return targetDiv.contains(range.startContainer) && targetDiv.contains(range.endContainer);
```
Comment (badasswp) [inline src/app.tsx:77]: @anandraj346 Let's put a space before the return statement like so:

```js
    const range = selection.getRangeAt(0);

    return targetDiv.contains(range.startContainer) && targetDiv.contains(range.endContainer);
```

**Review by badasswp — CHANGES_REQUESTED** [review]

File: src/app.tsx (hunk 2 above)
Comment (anandraj346) [reply src/app.tsx:227]: Ok sure

File: CHANGELOG.md (hunk 1 above)
Comment (anandraj346) [reply CHANGELOG.md:5]: Sure

**Review by badasswp — APPROVED** [review]
@anandraj346 This looks good. Thanks for spotting this. 👍
